src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java

```diff
@@ -0,0 +1,150 @@
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_creationTime_attribute_contains_the_creation_time_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_choices_attribute_contains_the_choices_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONArray jsonArray = vote.createChoicesJson(vote.choices);
+		
+		assertEquals(result.getJSONArray("choices").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canEndorse_attribute_contains_the_voters_endorse_choices_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canEndorse"), vote.canEndorse);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_countedAssurances_attribute_contains_the_counted_assurances_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONArray jsonArray = new JSONArray(vote.countedAssurances);
+		
+		assertEquals(result.get("countedAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_neededAssurances_attribute_contains_the_needed_assurances_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONArray jsonArray = new JSONArray(vote.neededAssurances);
+		
+		
+		assertEquals(result.get("neededAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_minEndorsements_attribute_contains_the_mininimum_endorsements_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("minEndorsements"), vote.minEndorsements);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_id_attribute_contains_the_id_of_the_vote() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("id"), vote.id);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canView_attribute_contains_the_voters_can_view_the_results() throws ReportedException {
```

Comment:
the canViev attribute contains whether the voters can view the results

Reply:
Fixed